openghg/standardise/_standardise.py

```diff
@@ -489,72 +508,48 @@ def standardise_flux(
-#     """Upload a file to the object store
-
-#     Args:
-#         filepath: Path of file to upload
-#     Returns:
-#         None
-#     """
-#     from gzip import compress
-#     import tempfile
-#     from openghg.objectstore import PAR
-#     from openghg.client import get_function_url, get_auth_key
-
-#     auth_key = get_auth_key()
-#     fn_url = get_function_url(fn_name="get_par")
-#     # First we need to get a PAR to write the data
-
-#     response = _post(url=fn_url, auth_key=auth_key)
-#     par_json = response.content
-
-#     par = PAR.from_json(json_str=par_json)
-#     # Get the URL to upload data to
-#     par_url = par.uri
-
-#     if filepath is not None and data is None:
-#         filepath = Path(filepath)
-#         MB = 1e6
-#         file_size = Path("somefile.txt").stat().st_size / MB
-
-#         mem_limit = 50  # MiB
-#         if file_size < mem_limit:
-#             # Read the file, compress it and send the data
-#             file_data = filepath.read_bytes()
-#             compressed_data = compress(data=file_data)
-#         else:
-#             tmp_dir = tempfile.TemporaryDirectory()
-#             compressed_filepath = Path(tmp_dir.name).joinpath(f"{filepath.name}.tar.gz")
-#             # Compress in place and then upload
-#             with tarfile.open(compressed_filepath, mode="w:gz") as tar:
-#                 tar.add(filepath)
-
-#             compressed_data = compressed_filepath.read_bytes()
-#     elif data is not None and filepath is None:
-#         compressed_data = gzip.compress(data)
-#     else:
-#         raise ValueError("Either filepath or data must be passed.")
-
-#     # Write the data to the object store
-#     put_response = _put(url=par_url, data=compressed_data, auth_key=auth_key)
-
-#     print(str(put_response))
+        if bucket is None:
+            bucket = get_writable_bucket(name=store)
+        return standardise(
+            data_type="emissions",
+            bucket=bucket,
+            filepath=filepath,
+            species=species,
+            source=source,
+            domain=domain,
+            database=database,
+            database_version=database_version,
+            model=model,
+            high_time_resolution=high_time_resolution,
+            period=period,
+            continuous=continuous,
+            chunks=chunks,
+            overwrite=overwrite,
+        )
+
+
+def standardise_from_binary_data(
+    bucket: str, data_type: str, binary_data: bytes, metadata: dict, file_metadata: dict, **kwargs: Any
+) -> Optional[dict]:
+    dclass = get_data_class(data_type)
+    with dclass(bucket) as dc:
+        result = dc.read_data(
+            binary_data=binary_data, metadata=metadata, file_metadata=file_metadata, **kwargs
+        )
+    return result
+
+
+def standardise_from_remote_source(bucket: str, data_type: str, data: dict, **kwargs: Any) -> Optional[dict]:
```

Comment:
Missing doc string here (and including that would make it easier to understand what this new function is doing)

Reply:
I removed this function. It was meant to be a front end for `store_data`, but I've just changed that back to using the `ObsSurface` context manager. (Most likely I will remove that context manager when I made the Metastore class, so I'll have to change the call slightly then, but I guess that is better than having an out of place function in standardise.)